astarte-utils/AstarteGenericProducer.cpp

```diff
@@ -41,12 +41,25 @@ QHash<QByteArray, QVariant::Type> AstarteGenericProducer::mappingToType() const
     return m_mappingToType;
 }
 
-bool AstarteGenericProducer::sendData(const QVariant &value, const QByteArray &target,
-        const QDateTime &timestamp, const QVariantHash &metadata)
+QHash<QByteArray, QVariant::Type> AstarteGenericProducer::mappingToArrayType() const
+{
+    return m_mappingToArrayType;
+}
+
+bool validateTarget(const QByteArray &target)
 {
     if (!target.startsWith('/') || target.endsWith('/') || target.contains("//")
             || target.contains(";") || target.contains('\n') || target.contains('\r')
             || target.contains('+') || target.contains('#')) {
+        return false;
+    }
+    return true;
```

Comment:
no need for `if (true) return true`; the expression result is already a boolean.